dartsim/src/Base.hh

```diff
@@ -367,6 +367,31 @@ class Base : public Implements3d<FeatureList<Feature>>
     return id;
   }
 
+  private: static math::Inertiald DivideInertial(
+               const math::Inertiald &_wholeInertial, std::size_t _count)
+  {
+    math::Inertiald dividedInertial;
+    math::MassMatrix3d dividedMassMatrix;
+    dividedMassMatrix.SetMass(_wholeInertial.MassMatrix().Mass() /
+                              static_cast<double>(_count));
+    dividedMassMatrix.SetMoi(_wholeInertial.Moi() *
```

Comment:
you have to be careful here with the inertial rotations. Since you are calling `Inertial::SetPose` below, you should not set and get the `Moi` as a `Matrix3`, since that has the inertial pose rotations embedded within. I would recommend using `MassMatrix3` APIs like `SetDiagonalMoments` and `SetOffDiagonalMoments` or `SetInertiaMatrix`

Reply:
I've replaced it with `MassMatrix3d::Moi` (not Inertiald::Moi) which does not take poses into account in https://github.com/gazebosim/gz-physics/pull/367/commits/e772f869fa11ee06456dc7e5093b10e1827c2f7e. I could go with the APIs you recommended if this is still confusing/error prone. It's just less lines of code this way.